File: BotManager/BotLib/src/main/java/com/rkarp/botlib/BotConfig.java
package com.rkarp.botlib;

import net.rickiekarp.core.settings.Configuration;
import net.rickiekarp.core.settings.LoadSave;

import java.io.File;

public class BotConfig {
    @LoadSave
    public static String nodeBinary;
    public static String appiumBinary = File.separator+"appium"+File.separator+"build"+File.separator+"lib"+File.separator+"main.js";

    public static File getModulesDirFile() {
        return new File(Configuration.config.getConfigDirFile() + File.separator + "modules");
    }

    public static int APPIUM_LOG_LEVEL = 1;

    public static String DEVICE_NAME;
    public static String VERSION;
    public static String UDID;
}

File: projects/JavaFXApps/BotManager/BotLib/src/main/java/net/rickiekarp/botlib/BotConfig.kt
package net.rickiekarp.botlib

import net.rickiekarp.core.settings.Configuration
import net.rickiekarp.core.settings.LoadSave

import java.io.File

object BotConfig {
    @LoadSave
    var nodeBinary: String? = null
    var appiumBinary = File.separator + "appium" + File.separator + "build" + File.separator + "lib" + File.separator + "main.js"

    val modulesDirFile: File
        get() = File(Configuration.config.configDirFile.toString() + File.separator + "modules")

    var APPIUM_LOG_LEVEL = 1

    var DEVICE_NAME: String? = null
    var VERSION: String? = null
    var UDID: String? = null
}
